### ISSUE TITLE
getProperty/setProperty regression

### ISSUE THREAD

**Issue by Siddounet** [InsertKoinIO/koin#960]
**Is your feature request related to a problem? Please describe.**
In Koin 2.1.3, we were able to use setProperty and getProperty  (`Koin.kt`) with "any", example :
```
/**
 * Retrieve a property
 * @param key
 */
 fun <T> getProperty(key: String): T? {
    return _propertyRegistry.getProperty(key)
}

/**
 * Save a property
 * @param key
 * @param value
 */
 fun <T : Any> setProperty(key: String, value: T) {
    _propertyRegistry.saveProperty(key, value)
}
```

We moved to 2.2.0, and now we can't because set/getProperty value can only be Strings : 

```
/**
 * Retrieve a property
 * @param key
 */
fun getProperty(key: String): String? {
    return propertyRegistry.getProperty(key)
}

/**
 * Save a property
 * @param key
 * @param value
 */
fun setProperty(key: String, value: String) {
    propertyRegistry.saveProperty(key, value)
}
```

So, now we aren't able to set "any" as property. We were using property for passing navHost to navigators for example.

**Describe the solution you'd like**
Seems to be a regression. I think the "old" way using "any" was pretty good for some specific use cases. So, reimplementing or extending with above old way seems to be a good solution

**Describe alternatives you've considered**
We didn't found any "good" alternative. For now, we are using singletons to store/save those values but without koin :(

**Target Koin project**
Asap if possible :)

**Comment by arnaudgiuliani** [comment]
Ok, need to check how to revert to more open API 👍

**Comment by arnaudgiuliani** [comment]
fixed in 2.2.2
